rfcs/2021-02-23-6531-performance-testing.md

```diff
@@ -499,3 +504,203 @@ pursue [Run Vector Continuously](#run-vector-continuously) in the future but the
 long-tail issues.
+
+## Plan of Action
+
+In this section we break down the plan of action for the proposed future work,
+given in the previous section. There are three complementary efforts. They will
+be pursued in a roughly a concurrent manner, depending on how many people are
+available for the work. As an example, the `diagnostic` task "Report on Vector's
+Environment" will improve the test-harness work with regard to increasing
+reliability but is not a blocking task. The criterion "Discover Tolerances" will
+help establish a baseline for test-harness tests introduced in "Expand Test
+Coverage" but we might also take some arbitrary commit as "base" for new tests.
+
+### Ad Hoc Sprint
+
+Performance work is a matter of being familiar with the code base under
+examination, it's goals and the context in which it runs. I'd like for everyone
+working on Vector performance as a full-time concern to have an initial ramp
+sprint.
+
+Completion criteria: My aim in my own ramp sprint will be to deepen my
+understanding of the code base by pursuing low-hanging fruit present in
+flamegraph traces of the project for representative workload, see test-harness
+section below, pair with people on their PRs and stub out documentation notes on
+performance work in Vector.
+
+### Criterion
+
+Our criterion benchmarks establish non-integrated and partially-integrated
+performance profiles for Vector components. Our existing strategy here is ideal
+and we intend no major changes.
+
+#### Push Into the Critical PR Path
+
+Our existing criterion benchmarks do act as a check on our PRs but are not
+mandatory. The feedback loop today is very long, roughly an hour to two hours
+and would provide significant friction for contributors if the check were
+mandatory presently. We must reduce the total time the criterion check takes and
+then, once done, push the criterion benchmarks into the critical path for PRs.
+
+Completion criteria: Without reducing the scope or reliability of our criterion
+benchmarks we intend to complete the full run in a reasonable amount of time,
+say 10 minutes. This implies giving more hardware to the CI process tasked with
+benchmarking and splitting the load across multiple machines.
+
+#### Expand Coverage
+
+Our existing approach to criterion is in a good state. We will be well-served by
+expanding this coverage. At a minimum every source/sink/transform must have a
+benchmark that establishes its bytes-per-second throughput. Additional
+benchmarks that help us determine whether regressions have been introduce and
+where are, of course, for the common good. The source/sink/transforms should be
+tackled in priority order.
+
+Completion criteria: Once every source/sink/transform is covered with criterion
+benchmarks to the standard where we can declare its bytes-per-second throughput
+and guard against regressions in the same this task will be completed.
+
+#### Discover Tolerances
```

Comment:
I really like this 👍

Reply:
This is something we're pretty close to having in hand already, I'll point out.